src/tests/unit/test_get_test_id.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from functools import cached_property
+from typing import cast
+
+import pytest
+
+from kontrol.foundry import Foundry
+
+
+class FoundryMock:
+
+    @cached_property
+    def all_tests(self) -> list[str]:
+        return [
+            'test%AssertTest.checkFail_assert_false()',
+            'test%AssertTest.test_assert_false()',
+            'test%AssertTest.test_assert_true()',
+            'test%AssertTest.testFail_assert_true()',
+        ]
+
+    @cached_property
+    def all_non_tests(self) -> list[str]:
+        return ['test%AssertTest.setUp()']
+
+    # NOTE: Hardcoded list of proofs used to mock Foundry.proof_ids_with_test()
+    @cached_property
+    def proof_ids(self) -> list[str]:
+        return [
+            'test%AssertTest.checkFail_assert_false():0',
+            'test%AssertTest.test_assert_false():0',
+            'test%AssertTest.test_assert_true():0',
+            'test%AssertTest.testFail_assert_true():0',
+            'test%AssertTest.setUp():0',
+            'test%AssertTest.setUp():1',
+        ]
+
+    def proof_ids_with_test(self, test: str, version: int | None = None) -> list[str]:
+        return Foundry.filter_proof_ids(self.proof_ids, test, version)
+
+    # NOTE: Mocked to call directly Foundry.matching_tests() according to the implementation from foundry.py.
+    # NOTE: This way we don't mock both Foundry.matching_sigs() and Foundry.matching_tests().
+    def matching_sigs(self, test: str) -> list[str]:
+        foundry = cast('Foundry', self)
+        return Foundry.matching_tests(foundry, [test])
+
+    def get_test_id(self, test: str, version: int | None) -> str:
+        foundry = cast('Foundry', self)
+        return Foundry.get_test_id(foundry, test, version)
+
+    # NOTE: hardcoded method required to mock Foundry.get_test_id().
+    # NOTE: It finds 1 as the highest version of any proof.
+    def resolve_proof_version(
+        self,
+        test: str,
+        reinit: bool,
+        user_specified_version: int | None,
+    ) -> int:
+        return 1
+
+
+TEST_ID_DATA = [
+    (
+        'with_version',
+        'AssertTest.setUp()',
+        0,
+        'test%AssertTest.setUp():0',
+    ),
+    (
+        'without_version',
+        'AssertTest.setUp()',
+        None,
+        'test%AssertTest.setUp():1',
+    ),
+    (
+        'proof_not_found',
+        'AssertTest.setUp()',
+        3,
+        'Found\\ no\\ matching\\ proofs\\ for\\ AssertTest\\.setUp\\(\\):3\\.',
+    ),
+    (
+        'multiple_matches',
+        'AssertTest.test_assert',
+        0,
+        'Found\\ 2\\ matching\\ proofs\\ for\\ AssertTest\\.test_assert:0\\.',
```

Comment:
Here I'm partially matching the error message because `Foundry.filter_proof_ids(self.proof_ids, test, version)` doesn't always return the elements in the same order.

Reply:
That's fine, though it might be a good idea to sort the results of `Foundry.filter_proof_ids`.